pkg/controller/hostpathprovisioner/controller.go

```diff
@@ -316,14 +325,24 @@ func (r *ReconcileHostPathProvisioner) Reconcile(context context.Context, reques
 			return reconcile.Result{}, err
 		}
 
-		cr.SetFinalizers(nil)
-
-		// Update CR
-		err = r.client.Update(context, cr)
+		if err := r.cleanDeployments(reqLogger, cr, namespace); err != nil {
+			return reconcile.Result{}, err
+		}
+		deployments, err := r.currentStoragePoolDeployments(reqLogger, cr, namespace)
 		if err != nil {
-			reqLogger.Error(err, "Unable to remove finalizer from CR")
 			return reconcile.Result{}, err
 		}
+		reqLogger.Info("Number of deployments still active", "count", len(deployments))
+		if len(deployments) == 0 {
+			cr.SetFinalizers(nil)
```

Comment:
What if there are other finalizers?  Shouldn't clobber

Reply:
Nobody else should be setting finalizers on my object. If they are I should clobber them.

Reply:
That’s not the way it works.  What if there is an Uber operator that creates the hpp cr.  Not your object

Reply:
Also addFinalizer should be changed to not do this: obj.SetFinalizers([]string{hppFinalizer})

Reply:
Fixed